elastic/kibana
# [Uptime] [Synthetics Integration] fall back to checking verification mode to determine if tls is enabled

## Summary

Fixes https://github.com/elastic/kibana/issues/115009

The Elastic Synthetics integration must provide backwards compatibility for integration policies created in earlier versions, because Fleet will not ship with auto upgrading package policies for the Elastic Synthetics integration until 8.1.0.

To support this backwards compatibility, https://github.com/elastic/kibana/pull/114914 was added to ensure that older policies are able to be upgraded to the most recent package version when editing. This resolves 99% of the compatibility issues.

However, we also changed the way we handle determining whether or not TLS is enabled during the package policy edit flow. This PR resolves those issues.

In 7.16, we added new logic to put a key onto the integration policy to determine if TLS had been enabled. This new key is used exclusively by the UI, not heartbeat, in order to render the TLS toggle as checked in the edit integration flow. That key is `__ui.is_tls_enabled` and `__ui.is_zip_url_tls_enabled` for normal http/tcp tls and for Zip URL tls, respectively.

However, policies created earlier than than 7.16 will not have this key. In order to support backwards compatibility, we check if `ssl.verification_mode` has been added the policy as a back up.

In reality, adding these new `__ui` keys for tls was not necessary and were added prematurely. The truth presence of `ssl.verification_mode` is sufficient way to check if tls has been abled for that policy, because `ssl.verification_mode` is always defined on every policy that has `ssl` enabled.

I don't feel comfortable removing the new `__ui.is_tls_enabled` and `__ui.is_zip_url_tls_enabled` after feature freeze, as it's working as expected for new policies. However, we should consider removing this in 8.0.0, as it's extra, unnecessary logic.


### Testing

1. Pull down this branch
2. Change your Kibana port to 5602
3. Start Kibana
4. Run `elastic-package stack up --version 8.0.0-SNAPSHOT -v`
5. Once elastic-package is up, run `curl -u elastic:changeme -X POST "http://localhost:9200/_security/user/kibana_system/_password?pretty" -H 'Content-Type: application/json' -d' { "password" : "changeme" } '`
6. Login to `localhost:5602`
7. Navigate directly to the synthetics integration on an old, pre-7.16 policy version by visiting `http://localhost:5602/axi/app/integrations/detail/synthetics-0.3.0/overview`
8. Create an HTTP monitor and enable TLS
9. Go to edit that HTTP monitor
10. Verify that TLS is enabled within the edit flow
<img width="824" alt="Screen Shot 2021-11-02 at 11 28 00 AM" src="https://user-images.githubusercontent.com/11356435/139910497-ea37b802-ac80-44ef-9d79-f94f91d831f1.png">
